File: js/using-webhooks/webhooks_validate_signature.js
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
const signatureHelper = require('@kontent-ai/webhook-helper');

// Generates a payload hash and compares it to the payload signature
const isValidSignature = (req, secret) => {
  return signatureHelper.isValidSignatureFromString(
    req.body, // Use raw body data from the request, i.e., by using body-parser
    secret,
    req.headers['x-kontent-ai-signature']
  );
};

File: net/using-webhooks/webhooks_validate_signature.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.AspNetCore.Webhooks;
using Microsoft.AspNetCore.Builder;

// Validates the 'X-Kontent-ai-Signature' header against the raw webhook payload.
// Rejects requests to /webhooks that Kontent.ai did not sign with the webhook's secret.
// Requires the Kontent.Ai.AspNetCore package; the secret goes to "WebhookOptions:Secret" in appsettings.json.
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

app.UseWebhookSignatureValidator(
    context => context.Request.Path.StartsWithSegments("/webhooks", StringComparison.OrdinalIgnoreCase),
    builder.Configuration.GetSection(nameof(WebhookOptions)));